src/math/exp_fcn_hifi.c

```diff
@@ -110,8 +125,52 @@ static void mul_s64(ae_int64 in_0, ae_int64 in_1, ae_int64 *ptroutbitshi, ae_int
 	producthi = AE_ADD64(producthihi, carry);
 
 	producthi = AE_ADD64(producthi, product_hl_lh_h);
-
 	*ptroutbitslo = productlolo;
+#elif SOFM_EXPONENTIAL_HIFI3 == 1
+
+	ae_int64 producthi_1c;
+	ae_int64 producthi_2c;
+	ae_int64 productlo_2c;
+
+	ae_int64 s0 = AE_SRLI64(in_0, 63);
```

Comment:
AE_SRAI64() should be better?

Reply:
Why would sign extension be preferred over zero? I saw that several test samples were failing.

Reply:
There is no need to continue the entering sign, implying that the requirement is to take absolute value.

According to the documentation, there is a distinction between `AE_SRLI64` and `AE_SRAI64`.
AE_SRLI64 is a 64-bit logical (`zero-extending`) right shift of AE_DR register d0 by immediate, with result placed in d.
```
proto AE_SRLI64 {out ae_int64 d, in ae_int64 d0, in immediate sa} {} {
    AE_SRLI64 d, d0, sa;
}
```

AE_SRAI64 is a 64-bit arithmetic (`sign-extending`) right shift of AE_DR register d0 by immediate, with result placed in d.

```
proto AE_SRAI64 {out ae_int64 d, in ae_int64 d0, in immediate sa} {} {
    AE_SRAI64 d, d0, sa;
}
```